docs/arcadia-users-group/20230228-intro-to-python-1/lesson.md

```diff
@@ -0,0 +1,548 @@
+
+---
+## 3. Using variables
+
+You assign variables in Python using the `variable = value` syntax. You can perform operations on a mix of hard-coded values and variables interchangeably. This is very useful when you expect to use a variable in many different places in your script - Python keeps track of its value so you don't have to.
+
+
+```python
+a = 1
+b = 6 + a
+c = a * b
+
+a + b + c
+```
+
+
+
+
+    15
+
+
+
+You can assign a different value to a variable later in your code, which replaces the original value. This can be useful, but it's also important to be careful not to overwrite variables you want to keep!
+
+
+```python
+a = 4
+
+a + b
+```
+
+
+
+
+    10
+
+
+
+---
+## 4. Built-in Python functions
+
+In addition to managing variables and values, Python also provides a variety of [built-in functions](https://docs.python.org/3/library/functions.html). Functions are "called" by chaining the function name with parentheses `()`. Parameters of that function can be typed between the parentheses to tell the function what to do.
+
+For example, the `print()` function takes any value and prints it as output.
+
+
+```python
+print('Hello, world!')
+
+print(a + b)
+```
+
+    Hello, world!
+    10
+
+
+We can call a function from within another function call. For example, the `type()` function gives the type of a Python value.
+
+
+```python
+print(type(4))
+print(type('Hello, world!'))
+```
+
+    <class 'int'>
+    <class 'str'>
+
+
+---
+## 5. Typecasting
+
+A bit earlier, we tried to evalute the expression `3 + '5'`. This failed because we were trying to add an `int` and a `str`, which isn't natively supported operation with the `+` operand.
+
+If we wanted to actually get the sum of 3 + 5 in this case, but `'5'` is given as a `str`, we can actually **force** `'5'` to be an `int` instead using typecasting, as below:
+
+
+```python
+3 + int('5')
+```
+
+
+
+
+    8
+
+
+
+You can see that now, instead of throwing a `TypeError`, the expression returns the expected value.  
+
+We typecast in Python by calling our data type as a function, e.g. `str()`, `int()`, `float()`, etc.  
+In some cases, you can't cast a value into another type; for example, running the code below throws a `ValueError`.
+
+
+```python
+int('banana')
+```
+
+
+    ---------------------------------------------------------------------------
+
+    ValueError                                Traceback (most recent call last)
+
+    Cell In[22], line 1
+    ----> 1 int('banana')
+
+
+    ValueError: invalid literal for int() with base 10: 'banana'
+
+
+This example is probably pretty intuitive, as it's hard to imagine how you would turn the string `'banana'` into an integer.  
+But there are plenty of cases where typecasting won't work, so be aware that this isn't a perfect solution.
+
+---
+## P1. Practice
```

Comment:
what's your plan for doing this live? I sometimes:
1. make a powerpoint that has the questions on it so learners can see them
2. have the questions in a separate tab so I can copy and paste themonto my screen and into a learner chat (like our slack channel or zoom chat)

lmk if you want to brainstorm this or practice it, it can be hard managing this!

Reply:
I can make a Google Slides document with the practice questions and examples!

Reply:
I've made a Google Slides document [here](https://docs.google.com/presentation/d/13HoulLCuudnAgTJ03yMZUw_LWtB805SLF5o-tZUXtFo/edit?usp=sharing) that covers the lesson - I'm still working on the animations.